pkg/server/commands/reverseexpand/reverse_expand.go

```diff
@@ -107,7 +107,7 @@ type UserRef struct {
 }
 
 type ReverseExpandQuery struct {
-	datastore               storage.RelationshipTupleReader
+	datastore               storage.Reader
```

Comment:
This expands the scope of what the `ReverseExpandQuery` needs. `ReverseExpandQuery` specifically needs a `RelationshipTupleReader` because it needs to be able to read tuples. It doesn't care about reading authorization model from storage.

If you want to pass a struct that implements the `storage.Reader` interface when constructing one of these queries than that would be fine because it implements the `RelationshipTupleReader` interface, but this type definition here is over scoped.